### ISSUE TITLE
[prefer-immutable-types] Doesn't take into consideration ignoreNamePattern with hash symbol

### ISSUE THREAD

**Issue by Clarchik** [eslint-functional/eslint-plugin-functional#910]
## Bug Report

Seems like rule does not work as intended.

### Expected behavior
Rule below should not produce error for variables which follows # symbol (TS private modifier)
```
'functional/prefer-immutable-types': [
      'error',
      {
        ignoreNamePattern: ['_', '#']
      }
    ],
```

### Actual behavior

Works for 

`_anyName variable`

But produces error for

`#anyName variable`

`Property should have a readonly modifier eslint [functional/prefer-immutable-types]`

### Steps to reproduce

Latest version (7.2.0) with the rule snippet above

**Comment by RebeccaStevens** [comment]
So the `#` symbol at the start of a private field's name isn't internally considered part of the name itself.
So `ignoreNamePattern: "^mutable"` would ignore `class Klass { #mutableFoo: string[] = []; }`

I could change this in the future to require a # for private fields, but this would be a breaking change.

You could use the `ignoreClasses: "fieldsOnly"` option, but this will ignore all class fields, not just private ones. Is this suitable for your needs?

**Comment by Clarchik** [comment]
@RebeccaStevens

> You could use the ignoreClasses: "fieldsOnly" option, but this will ignore all class fields, not just private ones. Is this suitable for your needs?
Not sure, need to test it.

I was playing around source code of the lib and found the solution for myself which works, but the question is - is it a breaking change?

Changing this

```
if (isIdentifier(node) || isPrivateIdentifier(node)) {
 mut_identifierText = node.name;
}
```

to this

```
if (isIdentifier(node)) {
 mut_identifierText = node.name;
} else if(isPrivateIdentifier(node)) {
 mut_identifierText = `#${node.name}`;
} 
```
I found that only hash symbol is recognized as `PrivateIdentifier` and adding # before var name - makes sense in my case.
What's your opinion on that?

**Comment by RebeccaStevens** [comment]
After thinking about this for awhile, I believe that yes, that should be the desired behavior. It is a breaking change though.
